OpenRA.Mods.Common/Traits/World/MusicPlaylist.cs

```diff
@@ -164,6 +164,14 @@ public void Play(MusicInfo music, Action onComplete)
 			Game.Sound.PlayMusicThen(music, onComplete);
 		}
 
+		public void SetBackgroundMusic(MusicInfo music)
+		{
+			currentSong = currentBackgroundSong = music;
+			CurrentSongIsBackground = music != null;
+
+			Play();
```

Comment:
As mentioned above, I don't think this should override the currently playing song.  If you want to force this behavior from a script, then you can instead do it by stopping the playing song.